### PR TITLE
Remove tasks from ExPyRe, only cores and nodes

### PR DESCRIPTION
**Pull request by bernstei**
Refactor knowledge of tasks out of ExPyRe and into workflow, so ExPyRe only needs to know about cores and nodes.

### CODE REVIEW

**Review by gelzinyte — COMMENTED** [review]
Just suggestions - I'm still running tests on our cluster

**Comment by gelzinyte** [comment]
@gabor1 @bernstei I had some suggestions for renaming some things, what do you think?

File: expyre/resources.py [hunk 1]
```diff
@@ -42,28 +44,27 @@ def find_nodes(self, partitions, exact_fit=True, partial_node=False):
         partitions: dict
```
[the remark this answers is not in the document]
Comment (bernstei) [reply expyre/resources.py:58]: done

File: expyre/resources.py [hunk 2]
```diff
@@ -4,32 +4,34 @@
...
-            max time for job in s (int) or time spec (str)
+        max_time: int / str
+            max time for job in sec (int) or time spec (str)
         n: (int, str)
-            int number of tasks or nodes to use and str 'tasks' or 'nodes'
-        ncores_per_task: int, default 1
-            cores per task, 0 for all cores in node
-        max_mem_per_task: int, str, default None
-            max mem per task in kB (int) or memory spec (str)
+            int number of cores or nodes to use and str 'cores' or 'nodes'
```
[the remark this answers is not in the document]
Comment (bernstei) [reply expyre/resources.py:32]: done

**Pull request merged**